exporter/awsemfexporter/datapoint.go

```diff
@@ -213,85 +219,184 @@ func (dps histogramDataPointSlice) IsStaleNaNInf(i int) (bool, pcommon.Map) {
+// - Sum is only assigned to the first split to ensure the total sum of the datapoints after aggregation is correct.
+// - Count is accumulated based on the bucket counts within each split.
 func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int, _ string, _ bool, _ *emfCalculators) ([]dataPoint, bool) {
 	metric := dps.ExponentialHistogramDataPointSlice.At(idx)
-
 	scale := metric.Scale()
 	base := math.Pow(2, math.Pow(2, float64(-scale)))
-	arrayValues := []float64{}
-	arrayCounts := []float64{}
+
 	var bucketBegin float64
 	var bucketEnd float64
+	const splitThreshold = 100
+	var currentBucketIndex = 0
+	var datapoints []dataPoint
+	var currentPositiveIndex = metric.Positive().BucketCounts().Len() - 1
+	var currentZeroIndex = 0
+	var currentNegativeIndex = 0
+	totalBucketLen := metric.Positive().BucketCounts().Len() + metric.Negative().BucketCounts().Len()
+	if metric.ZeroCount() > 0 {
+		totalBucketLen++
+	}
 
-	// Set mid-point of positive buckets in values/counts array.
-	positiveBuckets := metric.Positive()
-	positiveOffset := positiveBuckets.Offset()
-	positiveBucketCounts := positiveBuckets.BucketCounts()
-	bucketBegin = 0
-	bucketEnd = 0
-	for i := 0; i < positiveBucketCounts.Len(); i++ {
-		index := i + int(positiveOffset)
-		if bucketBegin == 0 {
-			bucketBegin = math.Pow(base, float64(index))
-		} else {
-			bucketBegin = bucketEnd
+	if totalBucketLen == 0 {
+		return []dataPoint{{
+			name: dps.metricName,
+			value: &cWMetricHistogram{
+				Values: []float64{},
+				Counts: []float64{},
+				Count:  metric.Count(),
+				Sum:    metric.Sum(),
+				Max:    metric.Max(),
+				Min:    metric.Min(),
+			},
+			labels:      createLabels(metric.Attributes()),
+			timestampMs: unixNanoToMilliseconds(metric.Timestamp()),
+		}}, true
+	}
+
+	for currentBucketIndex < totalBucketLen {
+		// Create a new dataPointSplit with a capacity of up to splitThreshold buckets
+		split := dataPointSplit{
+			cWMetricHistogram: &cWMetricHistogram{
+				Values: []float64{},
+				Counts: []float64{},
+				Max:    metric.Max(),
+				Min:    metric.Min(),
+				Count:  0,
+				Sum:    0,
+			},
+			length:   0,
+			capacity: splitThreshold,
 		}
-		bucketEnd = math.Pow(base, float64(index+1))
-		metricVal := (bucketBegin + bucketEnd) / 2
-		count := positiveBucketCounts.At(i)
-		if count > 0 {
-			arrayValues = append(arrayValues, metricVal)
-			arrayCounts = append(arrayCounts, float64(count))
+
+		// Only assign `Sum` if this is the first split to make sure the total sum of the datapoints after aggregation is correct.
+		if currentBucketIndex == 0 {
+			split.cWMetricHistogram.Sum = metric.Sum()
 		}
-	}
 
-	// Set count of zero bucket in values/counts array.
-	if metric.ZeroCount() > 0 {
-		arrayValues = append(arrayValues, 0)
-		arrayCounts = append(arrayCounts, float64(metric.ZeroCount()))
-	}
+		if totalBucketLen-currentBucketIndex < splitThreshold {
```

Comment:
Yes, I think you are correct. 
The `currentBucketIndex` is starting from 0, which means `currentBucketIndex` is not already counted. 
For example, totalBucketLen = 200,  and we already have 100 buckets in the first split,, then when it comes to the second split, `currentBucketIndex = 100`, where the 100th bucket haven't been processed. `totalBucketLen-currentBucketIndex = 200 - 100 = 100`, all the 100 buckets from `currentBucketIndex = 100` to `currentBucketIndex = 199` should be fit in one split. I will update the code to use `<=` here.

Reply:
Can we put `capacity` assignment BEFORE initialization, so that `capacity` becomes immutable, and we never bother changing its value after initialization.